packages/test/test-end-to-end-tests/src/test/dataStoreRetrieval.spec.ts

```diff
@@ -0,0 +1,175 @@
+					innerDataObjectFactory.type,
+				);
+				const innerDataStore = await requestFluidObject<ITestDataObject>(
+					innerDataStoreRouter,
+					"",
+				);
+
+				this.root.set(this.innerDataStoreKey, innerDataStore.handle);
+			}
+
+			protected async hasInitialized(): Promise<void> {
+				const innerDataStoreHandle = this.root.get<IFluidHandle<InnerDataObject>>(
+					this.innerDataStoreKey,
+				);
+				assert(innerDataStoreHandle !== undefined, "inner data store handle is missing");
+			}
+		}
+		const outerDataObjectFactory = new DataObjectFactory(
+			"OuterDataObject",
+			OuterDataObject,
+			[],
+			[],
+		);
+
+		let provider: ITestObjectProvider;
+		const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+			runtime.IFluidHandleContext.resolveHandle(request);
+
+		beforeEach(() => {
+			provider = getTestObjectProvider();
+		});
+
+		it("Requesting not bound data stores in detached container", async () => {
+			const request = provider.driver.createCreateNewRequest(provider.documentId);
+			const loader = provider.makeTestLoader();
+			const container = await loader.createDetachedContainer(provider.defaultCodeDetails);
+			// Get the default (root) dataStore from the detached container.
+			const mainDataStore = await requestFluidObject<ITestDataObject>(container, "/");
+
+			// Create another data store and bind it by adding its handle in the root data store's DDS.
+			const dataStore2 = await requestFluidObject<ITestDataObject>(
+				await mainDataStore._context.containerRuntime.createDataStore(TestDataObjectType),
+				"",
+			);
+			mainDataStore._root.set("dataStore2", dataStore2.handle);
+
+			// Request the new data store via the request API on the container.
+			const dataStore2Response = await container.request({
+				url: dataStore2.handle.absolutePath,
+			});
+			assert(
+				dataStore2Response.mimeType === "fluid/object" && dataStore2Response.status === 200,
+				"Unable to load bound data store in detached container",
+			);
+			await container.attach(request);
+		});
+
+		it("Requesting data store before outer data store completes initialization", async () => {
+			const containerRuntimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+				outerDataObjectFactory,
+				[
+					[outerDataObjectFactory.type, Promise.resolve(outerDataObjectFactory)],
+					[innerDataObjectFactory.type, Promise.resolve(innerDataObjectFactory)],
+				],
+				undefined,
+				[innerRequestHandler],
+			);
+			const request = provider.driver.createCreateNewRequest(provider.documentId);
+			const loader = provider.createLoader([
+				[provider.defaultCodeDetails, containerRuntimeFactory],
+			]);
+
+			const container = await loader.createDetachedContainer(provider.defaultCodeDetails);
+			// Get the outer dataStore from the detached container.
+			const outerDataStore = await requestFluidObject<ITestDataObject>(container, "/");
```

Comment:
Can we use entryPoint here?  `requestFluidObject` should be deprecated

Reply:
This is on the container and I don't think the test object provider passes initializeEntryPoint yet.